### DISCUSSION TITLE
Not understanding the final calculation results of R2D

### DISCUSSION THREAD

**Discussion by mazidong1** [NHERI-SimCenter/SimCenterCommon#221] [R2D] [answered]
Dear Experts，

As a beginner, I watched the introduction of rWHALE and R2D on the official website, and I thought I didn't miss any key operations.

Until I started to operate, I found that the final result of R2D's Example 2 was different every time I ran Example 2?

I carefully checked the THA and UQ modules and found that the seed values ​​were the same (which means that the outputs of the two modules should be unchanged, right?). 

I want to ask what is the root cause of the different results of Example 2 every time I run it? Is it in the DL module?

**Comment by jinyan1214** [comment, marked as answer]
Thanks for your questions. 

You are correct that the reason for different results in each run is in the DL module. Given the engineering demand parameters, the DL module generates damage measure and decision variable samples according to fragility curves. 

I will ask the developers of the DL module about how to set a seed for the DL so that results can be reproduced.

Best,

Jinyan

**Reply by mazidong1** [reply]
Many thanks, Jinyan.

**Comment by mazidong1** [comment]
I want to continue asking a question while you're still here. I was simulating a seismic disaster in a city and provided him with the seismic record of the nearest station (corrected), but according to the Python script, no hazard was generated. What does this have to do with?I don't think it's a problem with the format of my seismic motion. I know it's related to the THA module. What do you think? (Attachment: Seismic Ground Motion Record, Script of Errors Occurred)
![微信图片_20240911004443](https://github.com/user-attachments/assets/ab535f8d-49b2-4877-88ed-5c87a758043b)
![微信图片_20240911004619](https://github.com/user-attachments/assets/2a5d45fc-b765-4c23-8678-5fef7c02c734)

**Comment by jinyan1214** [comment]
How many points are there in your EventGrid.csv? If there is only 1 point, you can not find the 4 nearest girds point neighbors for each building. You may need to reduce the number of neighbors to 1.

**Comment by jinyan1214** [comment]
The error suggested that "Pattern" is not found in the "EVENT.json" in the
templatedir of your individual building. Please run one of the
built-in examples and compare the generated "EVENT.json" with the
"EVENT.json" generated in your own analysis. In the example "EVENT.json",
you should see a json key "pattern". If this json key doesn't exists in
your EVENT.json, try to look for errors from previous analysis steps.

The templatedir is in your
/documents/R2D/tmp.SimCenter/Results/Buildings/n.

**Reply by mazidong1** [reply]
Thank you Jinyan for your generous help. I just checked carefully and it is indeed my problem. I got the acceleration unit wrong.

Your suggestions are really helpful. I think I have asked enough questions today...So kind of you Jinyan ：）

**Comment by duyanan12** [comment]
Dear Experts,

I hope this message finds you well. My name is Yanan Du, a student at Xiamen University. I am currently in the early stages of researching earthquake resilience for Gulangyu Island and am interested in using the Simcenter R2D software for this purpose.

As a beginner with this software, I would greatly appreciate any guidance you can provide. Could you kindly advise if Simcenter R2D is applicable for research in China, especially for the Gulangyu region? Additionally, I would be grateful for any recommendations on how to get started with learning and applying the software effectively.

I sincerely hope to receive your advice and support, as I am eager to learn and apply this tool in my research.

Thank you very much for your time and assistance.

Best regards,
Yanan Du
Xiamen University

**Reply by jinyan1214** [reply]
Dear Yanan,

Thanks a lot for your interest in the software. The general workflow of R2D can be used to study earthquake resilience in China. However, there are several caveats that you need to be careful with.

1. The building inventory and transportation inventory generations function (through BRAILS) uses a data set from the US National Structure Inventory and is not applicable in China. You will need to create your own building inventory that complies with SimCenter's inventory format.

2. The earthquake rupture forecasting (ERF) model in the earthquake event generation toolbox primarily works for California. You can try to select OpenQuake as the ERF engine, but the interface between OpenQuake and R2D is not maintained very well due to the fast updates of both software. If you run into any error on using OpenQuake, please let us know. We can help you resolve the error.

3. The built-in fragility functions are developed by US FEMA's HAZUS program. They may not apply to buildings in China as the building design practices can be very different. It is recommended that you use your own customized fragility functions. 

The resources to learn are primarily the [software documentation](https://nheri-simcenter.github.io/R2D-Documentation/) and the SimCenter videos on the [Youtube channel](https://www.youtube.com/@NHERIDesignSafe/videos). 

I hope you find the answers useful.

Best regards,

Jinyan

**Comment by duyanan12** [comment]
Dear Jinyan

I wanted to express my sincere gratitude for your detailed and helpful response. Your insights into adapting R2D to the context of China have been invaluable, and I greatly appreciate the guidance you've provided.

I will begin by creating a building inventory in the SimCenter format and exploring the use of OpenQuake for ERF. I am also eager to customize the fragility functions to better reflect our local building practices.

Your prompt and informative reply has given me the confidence to proceed with these adjustments, and I am optimistic that they will lead to successful implementation.

Thank you once again for your support. I will keep you updated on my progress and will certainly reach out if I encounter any issues requiring your expertise.

Wishing you a wonderful day ahead.

Best regards,
Yanan Du



At 2024-10-18 12:27:34, "Jinyan Zhao" ***@***.***> wrote:

Dear Yanan,

Thanks a lot for your interest in the software. The general workflow of R2D can be used to study earthquake resilience in China. However, there are several caveats that you need to be careful with.

The building inventory and transportation inventory generations function (through BRAILS) uses a data set from the US National Structure Inventory and is not applicable in China. You will need to create your own building inventory that complies with SimCenter's inventory format.

The earthquake rupture forecasting (ERF) model in the earthquake event generation toolbox primarily works for California. You can try to select OpenQuake as the ERF engine, but the interface between OpenQuake and R2D is not maintained very well due to the fast updates of both software. If you run into any error on using OpenQuake, please let us know. We can help you resolve the error.

The built-in fragility functions are developed by US FEMA's HAZUS program. They may not apply to buildings in China as the building design practices can be very different. It is recommended that you use your own customized fragility functions.

The resources to learn are primarily the software documentation and the SimCenter videos on the Youtube channel.

I hope you find the answers useful.

Best regards,

Jinyan
